docs/source/fastapi/index.rst

```diff
@@ -93,6 +93,46 @@ We also mark one of the endpoints as deprecated.
         )
     )
 
+Authentication
+--------------
+
+If you want protect some endpoints using ``FastAPIKwargs``,
+you can pass auth dependencies into ``FastAPIApp.add_api_route``.
+
+Example
+~~~~~~~
+
+In the following example we pass dependencies into ``FastAPIKwargs`` to protect endpoints with unsafe HTTP methods.
+
+.. code-block:: python
+
+    from fastapi import Depends, FastAPI
+    from fastapi.security import OAuth2PasswordBearer
+    from piccolo_api.fastapi.endpoints import FastAPIWrapper, FastAPIKwargs
+    from piccolo_api.crud.endpoints import PiccoloCRUD
+
+    from my_app.tables import Task
+
+    app = FastAPI()
+
+    oauth2_scheme = OAuth2PasswordBearer(tokenUrl="login")
+
+    FastAPIWrapper(
+        root_url="/task/",
+        fastapi_app=app,
+        piccolo_crud=PiccoloCRUD(
+            table=Task,
+            read_only=False,
+        ),
+        fastapi_kwargs=FastAPIKwargs(
+            all_routes={'tags': ['Task']},  # Added to all endpoints
+            post={"dependencies": [Depends(oauth2_scheme)]}, # protected route
+            put={"dependencies": [Depends(oauth2_scheme)]},  # protected route
+            patch={"dependencies": [Depends(oauth2_scheme)]},  # protected route
+            delete_single={"dependencies": [Depends(oauth2_scheme)]},  # protected route
+        )
+    )
+
```

Comment:
Do you think it would be useful to link to your example project here? For example:

```
See this `example project <https://github.com/sinisaos/headless-forum-fastapi>`_ for more details.
```

Because your project also includes the login endpoint etc.